galleries/examples/specialty_plots/ishikawa_diagram.py

```diff
@@ -0,0 +1,200 @@
+    None.
+
+    """
+    second_sections = []
+    third_sections = []
+    # Resize diagram to automatically scale in response to the number
+    # of problems in the input data.
+    if len(args) == 1 or len(args) == 2:
+        spine_length = (-2.1, 2)
+        head_pos = (2, 0)
+        tail_pos = ((-2.8, 0.8), (-2.8, -0.8), (-2.0, -0.01))
+        first_section = [1.6, 0.8]
+    elif len(args) == 3 or len(args) == 4:
+        spine_length = (-3.1, 3)
+        head_pos = (3, 0)
+        tail_pos = ((-3.8, 0.8), (-3.8, -0.8), (-3.0, -0.01))
+        first_section = [2.6, 1.8]
+        second_sections = [-0.4, -1.2]
+    else:  # num == 5 or 6
+        spine_length = (-4.1, 4)
+        head_pos = (4, 0)
+        tail_pos = ((-4.8, 0.8), (-4.8, -0.8), (-4.0, -0.01))
+        first_section = [3.5, 2.7]
+        second_sections = [1, 0.2]
+        third_sections = [-1.5, -2.3]
+
+    # Change the coordinates of the annotations on each loop.
+    for index, problem in enumerate(args):
+        top_row = True
+        cause_arrow_y = 1.7
+        if index % 2 != 0:  # Plot problems below the spine.
+            top_row = False
+            y_prob_angle = -16
+            cause_arrow_y = -1.7
+        else:  # Plot problems above the spine.
+            y_prob_angle = 16
+        # Plot the 3 sections in pairs along the main spine.
+        if index in (0, 1):
+            prob_arrow_x = first_section[0]
+            cause_arrow_x = first_section[1]
+        elif index in (2, 3):
+            prob_arrow_x = second_sections[0]
+            cause_arrow_x = second_sections[1]
+        else:
+            prob_arrow_x = third_sections[0]
+            cause_arrow_x = third_sections[1]
+        if index > 5:
+            raise ValueError(f'Maximum number of problems is 6, you have entered '
+                             f'{len(args)}')
+
+        # draw main spine
+        ax.plot(spine_length, [0, 0], color=COLOR, linewidth=2)
+        # draw fish head
+        ax.text(head_pos[0] + 0.1, head_pos[1] - 0.05, 'PROBLEM', fontsize=10,
+                color='white')
+        semicircle = Wedge(head_pos, 1, 270, 90, fc=COLOR)
+        ax.add_patch(semicircle)
+        # draw fishtail
+        triangle = Polygon(tail_pos, fc=COLOR)
+        ax.add_patch(triangle)
+
+        problems(problem, prob_arrow_x, 0, -12, y_prob_angle)
+        causes(problem, cause_arrow_x, cause_arrow_y, top=top_row)
+
+
+# Input data
+method = ['Method', ['Time consumption', 'Cost', 'Procedures',
+                     'Inefficient process', 'Sampling']]
+machine = ['Machine', ['Faulty equipment', 'Compatibility']]
+material = ['Material', ['Poor-quality input', 'Raw materials', 'Supplier',
+                         'Shortage']]
+measure = ['Measurement', ['Calibration', 'Performance', 'Wrong measurements']]
+env = ['Environment', ['Bad conditions']]
+people = ['People', ['Lack of training', 'Managers', 'Labor shortage',
+                     'Procedures', 'Sales strategy']]
```

Comment:
Indeed, you can make this a single dict and pass it to draw_body().

Reply:
There's no advantage to it. I chose lists over a dictionary because @QuLogic disapproved of that format [in the old PR.](https://github.com/matplotlib/matplotlib/pull/25248#pullrequestreview-1316508821)

Reply:
I have not followed the whole topic, but my impression was that the no-dict comment refers to passing all data around through all functions as dict.

A dict as the top-level structure defining the diagram is much more concise:

```
categories = {
    'Method': ['Time consumption', 'Cost', 'Procedures', 'Inefficient process', 'Sampling'],
    'Machine': ['Faulty equipment', 'Compatibility'],
    'Material': ['Poor-quality input', 'Raw materials', 'Supplier',  'Shortage'],
    'Measurement': ['Calibration', 'Performance', 'Wrong measurements'],
    'Environment': ['Bad conditions'],
    'People': ['Lack of training', 'Managers', 'Labor shortage', 'Procedures', 'Sales strategy'],
}
```